### PR TITLE
Configure Helm repository ++ Add spring boot example app chart

### PR DESCRIPTION
**Pull request by Sgitario**
The repository has been configured to serve the static helm index and chart files. To use it:

```console
$ helm repo add snowdrop http://snowdrop.github.io/helm
```

And confirm that the snowdrop repository is listed:

```console
$ helm repo list
NAME           	URL                               
snowdrop	    http://snowdrop.github.io/helm
```

## Available Charts in Repository

| Chart Name                     | Description | Configuration |
|--------------------------------|-------------| --------------|
| spring-boot-example-app        | Chart to be used by the Snowdrop Spring Boot examples | [values.yaml](repository/spring-boot-example-app/values.yaml) |

## Usage

To use one of the available charts, for example `spring-boot-example-template`, update or add the `requirements.yaml` file to register the dependency:

```yaml
dependencies:
  - name: spring-boot-example-app
    version: 0.0.1
    repository: http://snowdrop.github.io/helm
```

Now, download the dependencies using: 
```console
$ helm dependency update
```

Configure the template accordingly using the chart name as root element:

```yaml
spring-boot-example-app: # This is the chart name from the dependencies!
  name: rest-http
  version: 2.5.0-0-SNAPSHOT
  s2i:
    source:
      repo: https://github.com/snowdrop/rest-http-example
      ref: sb-2.5.x
```

And, install your helm:
```console
$ helm install rest-http .
```

## Development

To add a new chart in the repository, you need to follow the following steps:
1. Add the new chart under `repository/<new chart name>`
2. Update the [Makefile](Makefile) file to add the new repository name in the `CHARTS` array
3. Run `make release` from the root repository folder

## Testing

1. Expose the repository locally using Docker and chartmuseum:

```console
docker run --rm -u 0 -it -d -p 8080:8080 -e DEBUG=1 -e STORAGE=local -e STORAGE_LOCAL_ROOTDIR=/charts -v $(pwd)/charts:/charts chartmuseum/chartmuseum:latest
```

The helm repository should be now available at `http://localhost:8080`.

2. Add the local repository using `helm repo add local http://localhost:8080`. Verify that the local chart is in the local repository `helm search repo local`.
3. Finally, update the `requirements.yaml` file to register the dependency:

```yaml
dependencies:
  - name: spring-boot-example-app
    version: 0.0.1
    repository: http://localhost:8080 # Helm local repository
```
Now, use the chart as stated in the [usage section](#usage).

### CODE REVIEW

**Comment by Sgitario** [comment]
@cmoulliard this PR configures this repository as a Helm provider (exposed via GitHub pages).
I've tested the new `spring-boot-example-app` using https://github.com/snowdrop/rest-http-example locally (see testing instructions in the README.md) and it worked fine.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+release:
+	@for chart in $(CHARTS) ; do \
+		chartFolder=$(PWD)/repository/$$chart; \
+		chartVersion=`grep '^version:' $$chartFolder/Chart.yaml | awk '{print $2}'`; \
+		echo Chart $$chart - $$chartVersion ; \
+		helm package $$chartFolder -d $(DESTINATION_FOLDER) ; \
+	done
+
+	cd $(DESTINATION_FOLDER)
+	helm repo index --url http://snowdrop.github.io/helm --merge index.yaml .
```
Comment (cmoulliard) [inline Makefile:16]: Is the plan to publish the index.yml file under snowdrop/helm github repo ?

File: README.md [hunk 2]
```diff
@@ -1,8 +1,107 @@
...
+  * [Usage](#usage)
+  * [Development](#development)
+  * [Testing](#testing)
+* [Templates](#templates)
+  * [Role-Based Access Control](#rbac)
 
-## How to play
+# Introduction
+
+This repository contains Helm templates and charts used by other applications part of the Snowdrop team.
```
Comment (cmoulliard) [inline README.md]: To be reviewed please

File: README.md [hunk 3]
```diff
@@ -1,8 +1,107 @@
...
+
+# Repository
+
+The repository has been configured to serve the static helm index and chart files. To use it:
```
Comment (cmoulliard) [inline README.md]: `The repository has been configured` How ? Can you be more explicit ?

File: README.md [hunk 4]
```diff
@@ -1,8 +1,107 @@
...
+## Development
+
+To add a new chart in the repository, you need to follow the following steps:
+1. Add the new chart under `repository/<new chart name>`
+2. Update the [Makefile](Makefile) file to add the new repository name in the `CHARTS` array
+3. Run `make release` from the root repository folder
+
+## Testing
+
+1. Expose the repository locally using Docker and chartmuseum:
```
Comment (cmoulliard) [inline README.md]: What is chartmuseum and why do we have to use it ? Maybe add a link to the prohect chartmuseum ?

File: README.md [hunk 5]
```diff
@@ -1,8 +1,107 @@
...
+
+## Available Charts in Repository
+
+| Chart Name                     | Description | Configuration |
+|--------------------------------|-------------| --------------|
+| spring-boot-example-app        | Chart to be used by the Snowdrop Spring Boot examples | [values.yaml](repository/spring-boot-example-app/values.yaml) |
+
+## Usage
+
+To use one of the available charts, for example `spring-boot-example-template`, update or add the `requirements.yaml` file to register the dependency:
```
Comment (cmoulliard) [inline README.md]: Where is such a `requirements.yaml` file located ?

File: Makefile (hunk 1 above)
Comment (Sgitario) [reply Makefile:16]: Yes. This is also how the redhat helm repository works. 
We don't need to do anything special to publish it. Using Github Page (https://github.com/snowdrop/helm/settings/pages) automatically publish the file `index.yml` at the root project.

File: README.md (hunk 2 above)
Comment (Sgitario) [reply README.md]: What do you mean?

File: README.md (hunk 3 above)
Comment (Sgitario) [reply README.md]: Changed. Let me know if it looks clearer now.

File: README.md (hunk 4 above)
Comment (Sgitario) [reply README.md]: Changed. Let me know if it looks clearer now.

File: README.md (hunk 5 above)
Comment (Sgitario) [reply README.md]: Changed. Let me know if it looks clearer now.

**Comment by Sgitario** [comment]
@cmoulliard PR updated with some of your comments. Others need clarification. Let me know!

File: README.md (hunk 2 above)
Comment (cmoulliard) [reply README.md]: You should also include pat of the introduction that this project also:
- Serves as Helm index repository
- Contains  best practices to design Helm charts (eg. RBAC, ...)

Comment (Sgitario) [reply README.md]: Changed

File: README.md [hunk 6]
```diff
@@ -1,8 +1,125 @@
...
+  * [Usage](#usage)
+  * [Development](#development)
+  * [Testing](#testing)
+* [Templates](#templates)
+  * [Role-Based Access Control](#rbac)
 
-## How to play
+# Introduction
+
+This repository contains best practices to design Helm charts from templates (eg. [RBAC](#rbac), ...) and serves as Helm index repository at (http://snowdrop.github.io/helm)[http://snowdrop.github.io/helm/index.yaml]
```
Comment (cmoulliard) [inline README.md:15]: I will review this part ;-)

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**